### PR TITLE
Crud operations

### PR DESCRIPTION
**Pull request by Jia-A**
In this PR, I have added following features for Dot on Time pomodoro app.
 - Add a new task.
 - Delete a task.
 - Edit a task.

### CODE REVIEW

File: src/context/task-context.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+const TaskProvider = ( { children } ) => {
+    const [ tasks, setTasks ] = useState([]);
+    const [editing, setEditing] = useState(false);
+    const navigate = useNavigate();
+    const [ singleTask, setSingleTask ] = useState({
+        name : "",
+        duration : 0,
+        description : "",
+    })
```
Comment (shuklahimanshu81) [inline src/context/task-context.js:16]: More than 2 state, try using context

**Review by shuklahimanshu81 — APPROVED** [review]
Everything looks good. Code is clean and understandable. Good to merge.

File: src/context/task-context.js (hunk 1 above)
Comment (Jia-A) [reply src/context/task-context.js:16]: Do you mean reducer?

**Review by abhij1607 — APPROVED** [review]
Neat and clean code
Good logic used

Can be merged

File: src/context/task-context.js (hunk 1 above)
Comment (shuklahimanshu81) [reply src/context/task-context.js:16]: Yeah sorry I forgot.

**Pull request merged**